src/DDM.jl

```diff
@@ -125,230 +226,493 @@ function _large_time_pdf(u::T, z::T, K::Int) where {T<:Real}
 
 # Arguments
-- `dist`: model object for the Diffusion Decision Model. 
+- `dist`: model object for Diffusion Decision Model. 
+    
+method simulating the diffusion process: 
+_rand_rejection uses Tuerlinckx et al., 2001 rejection-based method for the general wiener process
+_rand_stochastic uses the stochastic Euler method to simulate the stochastic differential equation
+
+# References
+
+    Tuerlinckx, F., Maris, E., Ratcliff, R., & De Boeck, P. (2001). 
+    A comparison of four methods for simulating the diffusion process. 
+    Behavior Research Methods, Instruments, & Computers, 33, 443-456.
+
+    Converted from Rhddmjagsutils.R R script by Kianté Fernandez
+    See also https://github.com/kiante-fernandez/Rhddmjags.
 """
 function rand(rng::AbstractRNG, d::DDM)
     return _rand_rejection(rng, d)
+    # return _rand_stochastic(rng, d)
 end
 
-# Rejection-based Method for the Symmetric Wiener Process(Tuerlinckx et al., 2001 based on Lichters et al., 1995)
-# adapted from the RWiener R package, note, here σ = 0.1
-function _rand_rejection(rng::AbstractRNG, d::DDM)
-    (ν, α, τ, z) = params(d)
-    
-    ϵ = 1.0e-15
-
-    D = .005 # = 2*σ^2 => 1/200
-    zn = (z*α) / 10 # absolute bias!
-    αn = α / 10
-    νn = ν / 10
-
-    total_time = 0.0
-    start_pos = 0.0
-    Aupper = αn - zn
-    Alower = -zn
-    radius = min(abs(Aupper), abs(Alower))
-    λ = 0.0
-    F = 0.0
-    prob = 0.0
-
-    while true
-        if νn == 0
-            λ = (0.25D*π^2) / (radius^2)
-            F = 1.0
-            prob = .5            
-        else
-            λ = ((0.25*νn^2)/D) + ((0.25*D*π^2) / (radius^2))
-            F = (D*π) / (radius*νn)
-            F = F^2 / (1+F^2)
-            prob = exp((radius*νn)/D)
-            prob = prob / (1+prob)
-        end
+function _rand_rejection(rng::AbstractRNG, d::DDM; N::Int = 1)
+    (ν, α, τ, z, η, sz, st, σ) = params(d)
 
-        r = rand(rng)
-        dir = r < prob ? 1 : -1
-        l = -1.0
-        s1 = 0.0
-        s2 = 0.0
-
-        # Tuerlinckx et al. (2001; eq. 16)  
-        while s2 > l
-            s1 = rand(rng)
-            s2 = rand(rng)
-            tnew = 0.0
-            tδ = 0.0
-            uu = zero(Int)
-
-            while (abs(tδ) > ϵ) || (uu == 0)
-                uu += 1
-                tt = 2*uu + 1
-                tδ = tt * (uu % 2 == 0 ? 1 : -1) * (s1 ^ (F * tt^2))
-                tnew += tδ                              
-            end
+    if η == 0
+        η = 1e-16
+    end
 
-            l = 1 + (s1^(-F)) * tnew 
+    # Initialize output vectors
+    choice = fill(0, N)
+    rt = fill(0.0, N)
+
+    # Called sigma in 2001 paper
+    D = σ^2 / 2
+
+    # Program specifications
+    ϵ = eps()  # precision from 1.0 to next double-precision number
+    Δ = ϵ
+
+    for n in 1:N
+        r1 = randn(rng)
+        μ = ν + r1 * η
+        bb = z - sz / 2 + sz * rand(rng)
+        zz = bb * α
+        finish = 0
```

Comment:
The compiler is more clever now, but it still might be a good idea to initialize `totaltime` and `startpos` as a float (i.e., 0.0).

A minor stylistic request: change finish to  `finished = false`, and `while finish == 0` to `while !finished`  to make it easier to read